engine/access/rest/request_decorator.go

```diff
@@ -0,0 +1,140 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// a convenience wrapper around the http request to make it easy to read request query params
+type requestDecorator struct {
+	*http.Request
+	expandFields map[string]bool
+	selectFields []string
+}
+
+func newRequestDecorator(r *http.Request) *requestDecorator {
+	decoratedReq := &requestDecorator{
+		Request: r,
+	}
+
+	if expandFields, found := middleware.GetFieldsToExpand(r); found {
+		decoratedReq.expandFields = sliceToMap(expandFields)
+	}
+
+	if selectFields, found := middleware.GetFieldsToSelect(r); found {
+		decoratedReq.selectFields = selectFields
+	}
+
+	return decoratedReq
+}
+
+func sliceToMap(values []string) map[string]bool {
+	valueMap := make(map[string]bool, len(values))
+	for _, v := range values {
+		valueMap[v] = true
+	}
+	return valueMap
+}
+
+func (rd *requestDecorator) expands(field string) bool {
+	return rd.expandFields[field]
+}
+
+func (rd *requestDecorator) selects() []string {
+	return rd.selectFields
+}
+
+func (rd *requestDecorator) getVar(name string) string {
+	vars := mux.Vars(rd.Request)
+	return vars[name] // todo(sideninja) consider returning err if non-existing
+}
+
+func (rd *requestDecorator) getQueryParam(name string) string {
+	return rd.Request.URL.Query().Get(name) // todo(sideninja) consider returning err if non-existing
+}
+
+func (rd *requestDecorator) getQueryParams(name string) []string {
+	param := rd.Request.URL.Query().Get(name)
+	// currently, the swagger generated Go REST client is incorrectly doing a `fmt.Sprintf("%v", id)` for the id slice
```

Comment:
same comment here about supporting space delimiting vs dropping support

Reply:
not sure what you mean with this?

Reply:
I have context on this and agree - will add back the support for space delimited params.

Reply:
changed.